libtransact/src/context/manager/thread/mod.rs

```diff
@@ -138,18 +138,20 @@ impl ContextManagerThread {
                     break;
                 }
                 Err(err) => {
-                    error!("ContextManagerThread recv error: {}", err);
-                    break;
+                    return Err(ContextManagerThreadError::ThreadReceiveError(err));
                 }
             }
         }
+        Ok(())
     }
 
     pub fn start(mut self) -> std::thread::JoinHandle<()> {
         thread::Builder::new()
             .name(String::from("Thread-ContextManager"))
             .spawn(move || {
-                self.run();
+                if let Err(err) = self.run() {
+                    error!("ContextManagerThread ended due to error: {}", err);
```

Comment:
Generally, this error means all senders have been dropped, so it may mean that the parts of the system that send item to the background thread have also been shutdown and their resources cleaned up before a shutdown signal has been sent.